citrination_client/search/client.py

```diff
@@ -32,6 +32,19 @@ def _handle_response(self, response, failure_message=DEFAULT_FAILURE_MESSAGE):
 
         return super(SearchClient, self)._handle_response(response, failure_message)
 
+    def _validate_query_depth(self, pif_system_returning_query):
```

Comment:
This is totally fine. However, for the sake of learning about some other ways you could improve it...
- You could change the name to just `_validate_query`, and then use that for any validation that you might want to add in the future.
- You could rename either of `_validate_query` or `_execute_paginating_search` such that they have more consistent names. I would probably choose something like `_validate_search_query` and `_execute_search_query`, but the important thing is that they describe reality and are consistent, for the sake of the reader.
- You could consider moving the call to the `_validate_query` method to inside of the `_execute_query` method, so that any query is automatically validated, as opposed to having to call those things separately. There are reasonable concerns with that approach, though (namely, what if I wanted to execute a non validated query?).
- You could improve the wording of the error message to be more explicit, E.g. `Reduce either of from_index or size arguments such that their sum is below {MAX_QUERY_DEPTH}`